serialzy/serializers/onnx.py

```diff
@@ -0,0 +1,54 @@
+import inspect
+import os
+from pathlib import Path
+from typing import BinaryIO, Union, Type, Optional, Any
+
+from serialzy.serializers.base_model import (
+    ModelBaseSerializer,
+    deserialize_from_file,
+    serialize_to_file,
+    unpack_model_file
+)
+
+
+class ONNXSerializer(ModelBaseSerializer):
+    def __init__(self):
+        super().__init__("onnx", __name__)
+
+    def _types_filter(self, typ: Type) -> bool:
+        import onnx
+        return inspect.isclass(typ) and issubclass(typ, onnx.ModelProto)
+
+    def _serialize(self, obj: Any, dest: BinaryIO) -> None:
+        import onnx
+
+        if not isinstance(obj, onnx.ModelProto):
+            raise ValueError(f"Attempt to serialize not an ONNX model with {__name__}")
+
+        assert isinstance(obj, onnx.ModelProto)
+
+        def write_onnx(out_filename: str) -> None:
```

Comment:
Instead of closure, make it method, classmethod or staticmethod, I think.

Reply:
why?

Reply:
Because you need to create this closure every time you executing external method. It is unnecessary overhead.

Reply:
Oh, I understand that you mean. There is not a problem with inner method but with closure creation because of `obj` object

Reply:
`save_to_file` expects one-arg function

Reply:
ah ok, I missed serialize_to_file point, yep